browser/js/common/directives/videoplayer/videoplayer.js

```diff
@@ -1,3 +1,4 @@
+// SO LONG
```

Comment:
this directive is HUGE. perhaps we can enhance readability by abstracting the events in it into another factory.